core/matrix/dense.cpp

```diff
@@ -1427,6 +1428,21 @@ void Dense<ValueType>::get_imag(
 }
 
 
+template <typename ValueType>
+void Dense<ValueType>::add_scaled_identity_impl(const LinOp* const a,
+                                                const LinOp* const b)
+{
+    precision_dispatch_real_complex<ValueType>(
```

Comment:
it already did the conversion, I think

Reply:
Right, the new test still works if I remove the make_temporary_conversion. I guess it would only be needed if I added the possibility for scalar type to be something other than value type. I think I'll delete the last commit.